src/theory/strings/theory_strings.cpp

```diff
@@ -1147,8 +1251,40 @@ void TheoryStrings::eqNotifyNewClass(TNode t){
     }
     //we care about the length of this string
     registerTerm( t[0], 1 );
-  }else{
-    //getExtTheory()->registerTerm( t );
+    return;
+  }
+  else if (k == CONST_STRING)
+  {
+    EqcInfo* ei = getOrMakeEqcInfo(t);
+    ei->d_prefixC = t;
+    ei->d_suffixC = t;
+    return;
+  }
+  else if (k == STRING_CONCAT)
+  {
+    addPrefixToEqcInfo(t, t, t);
+  }
+}
+
+void TheoryStrings::addPrefixToEqcInfo(Node t, Node concat, Node eqc)
+{
+  EqcInfo* ei = nullptr;
+  // check each side
+  for (unsigned r = 0; r < 2; r++)
+  {
+    unsigned index = r == 0 ? 0 : concat.getNumChildren() - 1;
+    Node c = utils::getConstantComponent(concat[index]);
+    if (!c.isNull())
+    {
+      if (ei == nullptr)
+      {
+        ei = getOrMakeEqcInfo(eqc);
+      }
+      Trace("strings-eager-pconf-debug")
+          << "New term: " << concat << " for " << t << " with prefix " << c
+          << " (" << (r == 1) << ")" << std::endl;
+      setPendingConflict(ei->addPrefixConst(t, c, r == 1));
```

Comment:
Hm, if we use `setPendingConflict()` in this way, maybe we should rename it to `setPendingConflictIf()`

Reply:
Renaming to `setPendingConflictWhen`